ioc_impl.py

```diff
@@ -392,6 +392,12 @@ def addIOC(self, index):
                                            "Failed to set required parameters for new IOC!",
                                            QtWidgets.QMessageBox.Ok, QtWidgets.QMessageBox.Ok)
                 continue
+            if self.model.findid(name) is not None:
+                QtWidgets.QMessageBox.critical(None,
```

Comment:
is there an issue with setting the parent to something that's not `None`  (maybe `self`)?  In my past experience doing this opens up the widget to garbage collection, which we don't really control.

Reply:
Hmm... I always just reflexively put None as the parent for messageboxes, but indeed, perhaps I shouldn't?  That's a larger question... because they are *all* None everywhere here.

Reply:
I did notice that.  It's fine here, we can change it later on.  If it's not been causing problems here maybe we don't worry about it.

Reply:
I think `None` makes the window a top-level widget but I don't know what the consequences of this are in terms of garbage collection.